src/actions/asyncActions.ts

```diff
@@ -119,6 +126,58 @@ export const AsyncActions = {
             dispatch(Actions.updateRssPosts(posts));
         };
     },
+    downloadPrivatePostsFromContacts: (contactFeeds: ContactFeed[]): Thunk => {
+        return async (dispatch, getState) => {
+            const identity = getState().author.identity!;
+            const profile: PrivateProfile = {
+                name: getState().author.name,
+                image: getState().author.image,
+                identity,
+            };
+            const feedAddress = Swarm.makeFeedAddressFromPublicIdentity(identity);
+            const signDigest = (digest: number[]) => Swarm.signDigest(digest, identity);
+            const swarmApi = Swarm.makeApi(
+                feedAddress,
+                signDigest,
+                getState().settings.swarmGatewayAddress
+            );
+            const storage = new SwarmStorage(swarmApi);
+            const crypto: Crypto = {
+                ...makeNaclEncryption(),
+                signDigest,
+                deriveSharedKey: (publicKey: HexString) => deriveSharedKey(identity, {publicKey, address: ''}),
+                random: (length: number) => generateSecureRandom(length),
+            };
+            const previousPosts = getState().rssPosts;
+            const allPosts: Post[] = [];
+            for (const contactFeed of contactFeeds) {
+                if (contactFeed.contact != null) {
+                    const contact = contactFeed.contact;
+                    const context = await makePrivateSharingContextWithContact(
+                        profile,
+                        contact.identity,
+                        storage,
+                        crypto,
+                    );
+                    const updatedContext = await privateSync(context);
+                    const timelinePosts = listTimelinePosts(updatedContext.remoteTimeline);
+                    for (const post of timelinePosts) {
+                        const postWithAuthor = {
+                            ...post,
+                            author: {
+                                name: contact.name,
+                                uri: contactFeed.feedUrl,
+                                image: contact.image,
+                            },
+                        };
+                        allPosts.push(postWithAuthor);
+                    }
+                }
+            }
+            const posts = mergeUpdatedPosts(allPosts, previousPosts);
+            dispatch(Actions.updateRssPosts(posts));
```

Comment:
This action should be renamed

Reply:
This updates the `rssPosts` field in the AppState, hence the name.

What is your suggestion?